### PR TITLE
query by order status

### PR DESCRIPTION
**Pull request by anuva98**
Added query string for find_by_status which queries all orders by order_status. 
Added test cases.

### CODE REVIEW

File: service/models/order.py [hunk 1]
```diff
@@ -73,3 +77,22 @@ def find_by_name(cls, name):
...
+
+    @classmethod
+    def find_by_status(cls, order_status: str) -> list:
+        """Returns list of all Orders with the given order status
+        Args:
+            status (string): the status of orders you want
+        """
+       
+        if(order_status in Order_Status.list()):
+            print("hereeee")
```
Comment (rheachandok) [inline service/models/order.py:89]: Can remove debug print

File: tests/test_routes.py [hunk 2]
```diff
@@ -543,3 +543,32 @@ def test_update_item_item_not_found(self):
             content_type="application/json",
         )
         self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
+
+# ----------------------------------------------------------
```
Comment (rheachandok) [inline tests/test_routes.py:547]: Why are the tests in routes?

Comment (Hyxpillow) [reply tests/test_routes.py:547]: Since both model test cases and endpoint test cases are needed, "test_routes.py" contains the endpoint test cases here.

**Review by Hyxpillow — APPROVED** [review]
LGTM, But when we do  "GET /order?name=Alice&order_status=CREATED", we would get all Alice's orders, not only "CREATED" but also "IN_PROGRESS", etc. 
Anyway, it's OK here and I will implement multi-filtering in my upcoming PR.

**Comment by Hyxpillow** [comment]
Oh, there're some conflicts here, you may try and solve them.

File: service/models/order.py (hunk 1 above)
Comment (anuva98) [reply service/models/order.py:89]: oh shoot completely forgot

**Review by Hyxpillow — APPROVED** [review]
LGTM

**Review by Hyxpillow — CHANGES_REQUESTED** [review]
image
Oh, I got 3 test cases failed during "make test", you may try and fix them.

**Review by Hyxpillow — APPROVED** [review]
Looks good to me.

**Pull request merged**